docs/developer-docs/latest/development/using-providers.md

```diff
@@ -0,0 +1,203 @@
+:::
+
+::: tab Email
+
+```js
+module.exports = ({ env }) => ({
+  // ...
+  email: {
+    config: {
+      provider: 'sendgrid', // For community providers pass the full package name (e.g. provider: 'strapi-provider-email-mandrill')
+      providerOptions: {
+        apiKey: env('SENDGRID_API_KEY'),
+      },
+      settings: {
+        defaultFrom: 'juliasedefdjian@strapi.io',
+        defaultReplyTo: 'juliasedefdjian@strapi.io',
+        testAddress: 'juliasedefdjian@strapi.io',
+      },
+    },
+  },
+  // ...
+});
+```
+
+Keep in mind that:
+
+* When using a different provider per environment, specify the correct configuration in `./config/env/${yourEnvironment}/plugins.js` (See [Environments](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md)).
+* Only one email provider will be active at a time. If the email provider setting isn't picked up by Strapi, verify the `plugins.js` file is in the correct folder.
+* When testing the new email provider with those two email templates created during strapi setup, the _shipper email_ on the template defaults to `no-reply@strapi.io` and needs to be updated according to your email provider, otherwise it will fail the test (See [Configure templates locally](/user-docs/latest/settings/configuring-users-permissions-plugin-settings.md#configuring-email-templates)).
+
+:::
+
+::::
+
+### Configuration per environment
+
+When configuring your provider you might want to change the configuration based on the `NODE_ENV` environment variable or use environment specific credentials.
+
+You can set a specific configuration in the `./config/env/{env}/plugins.js` configuration file and it will be used to overwrite the default configuration.
+
+## Creating providers
+
+To implement your own custom provider you must [create a Node.js module](https://docs.npmjs.com/creating-node-js-modules).
+
+The interface that must be exported depends on the plugin you are developing the provider for. Below are templates for the Upload and Email plugins:
+
+:::: tabs card
+
+::: tab Upload
+
+```js
+module.exports = {
+  init(providerOptions) {
+    // init your provider if necessary
+
+    return {
+      upload(file) {
+        // upload the file in the provider
+        // file content is accessible by `file.buffer`
+      },
+      uploadStream(file) {
+        // upload the file in the provider
+        // file content is accessible by `file.stream`
+      },
+      delete(file) {
+        // delete the file in the provider
+      },
+    };
+  },
+};
+```
+
```

Comment:
OK. Weird. For me, it was only showing the `upload` tab.

Reply:
You indeed to keep the closing markup. Otherwise, the tab is not rendered at all.
![Screenshot 2022-07-12 at 15 58 47](https://user-images.githubusercontent.com/4233866/178508358-506a8e0a-3180-49c1-ab37-69737b1fc8c1.png)